### ISSUE TITLE
[HOLD for payment 2023-07-20] [$4000]  Emoji reaction - Emoji reaction names are not in Spanish language 

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#16086]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Access staging.new.expensify.com
2. Sign into a valid account in Spanish language
3. Navigate to a conversation
4. Enter any data
5. Click on the emoji reaction
6. Hover over emoji reaction

## Expected Result:
The name of the emoji reactions must be in Spanish language if the selected language is Spanish

## Actual Result:
Emoji reaction names are not in Spanish language 

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.87.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![emoji spanish](https://user-images.githubusercontent.com/93399543/226060486-82ad95c3-bd65-404b-a3b0-150eb2cdaa13.png)
![emoji spanish2](https://user-images.githubusercontent.com/93399543/226060488-ccae7842-c0d5-4563-8e26-3ff3312734ab.png)

https://user-images.githubusercontent.com/93399543/226060491-f45bc11d-d168-4b11-a972-13c6451526df.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause -Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01dd45c43c01c59e34</li>
        <li>Upwork Job ID: 1650625983786610688</li>
        <li>Last Price Increase: 2023-06-13</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @tjferriss (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by tjferriss** [comment]
Confirming whether or not this is a duplicate of https://github.com/Expensify/App/issues/16051

**Comment by tjferriss** [comment]
Still waiting on a response for the above comment.

**Comment by stitesExpensify** [comment]
@tjferriss these are definitely different issues.

That said, I'm really not sure how to actually fix this problem. It will be a large change either way and we should definitely have an internal discussion about how to do it. [This is our file right now](https://github.com/Expensify/App/blob/cec24816e63f6de82d195bace5c80883ff26ba00/assets/emojis.js) and it only supports english. We will probably have to make an entire copy of that file and have it all translated to spanish (or find an existing list) which is a big undertaking. Then when we display things to others we will need to check against all language files instead of just USA.

I think we should make this monthly and have a discussion when we have time

**Comment by MelvinBot** [comment]
@tjferriss @stitesExpensify this issue was created 2 weeks ago. Are we close to a solution? Let's make sure we're treating this as a top priority. Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by tjferriss** [comment]
Pretty sure we're keeping this monthly for now, MelvinBot.

**Comment by stitesExpensify** [comment]
Removing bug label which should shut melvin up

**Comment by JmillsExpensify** [comment]
@iwiznia Curious for your take on this issue. I think we have to make this list internationalized before too long. Otherwise more and more parts of the product that aren't internalized will start to creep in.

**Comment by iwiznia** [comment]
Oh never even thought about that... I checked slack and was 🤯 that they translated all emojis:
<img width="580" alt="image" src="https://user-images.githubusercontent.com/521248/233724453-2b4946db-a554-4188-bcca-bf902081f131.png">

We can fix this if we feel it is important enough, though I am not eager to translate all the emoji names...

**Comment by JmillsExpensify** [comment]
Ha well I'm not eager either but I think we have to be committed to internationalizing the app in a consistent way.

**Comment by stitesExpensify** [comment]
Hopefully someone will be able to find an existing list instead of manually doing it ourselves 😅 

The other part of this change will be that we need to either:
1. Always search all emoji lists for matches (currently just spanish and english)
or
2. Change our search logic to know which list to search based on your selected language

IMO 2 is the way to go here. Either way though, I think that we should make this external. Thoughts?

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01dd45c43c01c59e34

**Comment by MelvinBot** [comment]
Current assignee @tjferriss is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @0xmiroslav (`External`)

**Comment by MelvinBot** [comment]
Current assignee @stitesExpensify is eligible for the External assigner, not assigning anyone new.

**Comment by gulfam110** [comment]
Here is the possible solution for this
Define a list of Spanish language emojis: Create a list or dictionary of Spanish language emojis that you want to search for in the text. Each emoji should be assigned a name and its corresponding Unicode code.

Write a regular expression: Use regular expressions to search for the Unicode codes of the Spanish language emojis in the text. A regular expression is a pattern of characters that can match specific strings of text.

Define a function: Write a function that takes a text input and searches for any matches of the regular expression pattern defined in Step 2. The function should return the names of the Spanish language emojis that were found in the text.

Test the function: Test the function on different input texts to ensure that it can correctly identify Spanish language emojis.

I hope this information finds Expensify well to get it started

Thanks,

**Comment by MelvinBot** [comment]
📣 @gulfam110! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by gulfam110** [comment]
Contributor details
My Expensify account email: gulfam.luqon@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01721019a355863460

**Comment by gulfam110** [comment]
@MelvinBot when would I hear back for this job?
As submitted the solution

**Comment by gulfam110** [comment]
Contributor details
Your Expensify account email: gulfam.luqon@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01721019a355863460

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by stitesExpensify** [comment]
Doubling the bounty

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $2000

**Comment by romulo114** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Emoji reaction names are not translated.

### What is the root cause of that problem?
We didn't add translation to the emoji-reactions as you can see below.

https://github.com/Expensify/App/blob/ac3f3c4658c359c10a3536c9ab0a64a0060300fe/src/components/Reactions/MiniQuickEmojiReactions.js#L67-L72

https://github.com/Expensify/App/blob/ac3f3c4658c359c10a3536c9ab0a64a0060300fe/src/components/Reactions/QuickEmojiReactions/BaseQuickEmojiReactions.js#L53-L61

### What changes do you think we should make in order to solve the problem?

We need to add translations to the emoji-reactions.
First, we need to add translations to `en.js` and `es.js` for emoji names: '+1', 'heart', 'joy', and 'fire'.

![image](https://user-images.githubusercontent.com/117526965/235646847-8b7a75b6-c477-4904-8211-dc5ef0338d3f.png)

Second, add the translation to both components by replacing `:${emoji.name}:` with `:${props.translate('emojiReactions.' + emoji.name)}:`

Also, we need to add `withLocalize` to the `BaseQuickEmojiReactions` component as in 
https://github.com/Expensify/App/blob/ac3f3c4658c359c10a3536c9ab0a64a0060300fe/src/components/Reactions/MiniQuickEmojiReactions.js#L103-L110

These updates fix this issue for both context menu and emoji reaction toolbar.

<details>
    <summary>Result</summary>
    

https://user-images.githubusercontent.com/117526965/235646405-ee9e857b-1c82-4fbe-9f84-966a4300415c.mp4

</details>

### What alternative solutions did you explore? (Optional)
NA

**Comment by JmillsExpensify** [comment]
@0xmiroslav Thoughts on the above proposal?

**Comment by JmillsExpensify** [comment]
Great question! Yes, I think we should.

**Comment by iwiznia** [comment]
> @JmillsExpensify since this will be large project, let's increase the bounty more to get proposals.

I don't think this is that large. $2000 seems reasonable and if it is not, the price will go up on its own by following our process.

**Comment by tjferriss** [comment]
@romulo114 are you able to update your proposal based on @0xmiroslav suggestion? https://github.com/Expensify/App/issues/16086#issuecomment-1536538352

**Comment by romulo114** [comment]
> @romulo114 are you able to update your proposal based on @0xmiroslav suggestion? [#16086 (comment)](https://github.com/Expensify/App/issues/16086#issuecomment-1536538352)

Yeah. I'm doing some research work.

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Emoji tooltips and search doesn't support Spanish.

### What is the root cause of that problem?

We don't have any mechanism to support multi-language for emojis.

### What changes do you think we should make in order to solve the problem?

#### Purpose: 
We have to support multi-language for emoji names and keywords. For example, :heart: emoji has a name `heart`, short code `:heart:`, and keywords `heart, red heart`. We don't need to translate short codes, but support multi-language for names and keywords.

We can use Spanish codes defined [here](https://emojiterra.com/es/lista/). As you can see from the pages, each emoji has its detail page and the detail page has keywords, short codes, and name.

In order to do this, we should do:
1. Extract a collection of emojis in `emojis.js` from the above URL. Extracted data would look like below or similar:
```
    {
        short: 'grinning',
        code: '😀',
        name: {
          en: 'grinning face',
          es: 'cara sonriendo',
        },
        keywords: {
          en: [
            'smile',
            'happy',
            'grinning',
            'face',
            'grin',
          ],
          es: [
            'cara',
            'cara sonriendo',
            'sonrisa',
            'feliz',
            'divertido'
          ]
        }
    }
```

2. Create 2 versions of EmojiTree in [EmojiTrie.js](https://github.com/Expensify/App/blob/32662bf639971d0a15b59d13148b83ae271ca716/src/libs/EmojiTrie.js#L8).

3. Add an extra parameter named lang with default value `en` to the [suggestEmojis](https://github.com/Expensify/App/blob/32662bf639971d0a15b59d13148b83ae271ca716/src/libs/EmojiUtils.js#L246) function.

```
function suggestEmojis(text, limit = 5, lang = 'en')
```

We can use the emoji tree for the specific `lang` parameter.

4. Add translations to the emoji reaction as described [here](https://github.com/Expensify/App/issues/16086#issuecomment-1531262203). But I strongly suggest to keep the short codes (such as `:heart:`) in English. I think we should add translations only for names and keywords as other popular applications do. We can add tooltip like `red heart (:heart:)` or `corazón rojo (:heart:)` in Spanish.

5. We might need to fix bugs in other emoji functions in EmojiUtils.js

<details>
    <summary>Result</summary>

</details>

### What alternative solutions did you explore? (Optional)

**Comment by s-alves10** [comment]
@0xmiroslav 
If that is a must, we can extend our emoji data like below.
```
    short: { en: 'grinning', es: 'sonriendo' }
```

And if the current locale is `es`, we need to add short codes for both `es` and `en` to keywords.

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/16086#issuecomment-1545263685)

**Comment by jayeshmangwani** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Emoji names are not translated into Spanish throughout the app.

### What is the root cause of that problem?
We use static English names/keywords for emojis for the emojis from assets/emojis.js file, e.g, left_speech_bubble here is static in name and keywords
https://github.com/Expensify/App/blob/5743b159f7ae44fb792f87d860b161c6b5c7bbf1/assets/emojis.js#L849-L853

### What changes do you think we should make in order to solve the problem?
I have tried a few approaches to make this work, but below adding that, I think that it will work without breaking existing reacted emojis

**1.** 
Need to create a new translation key `emojis` to [en.js](https://github.com/Expensify/App/blob/main/src/languages/en.js) and [es.js](https://github.com/Expensify/App/blob/main/src/languages/es.js) files where we are going to add all the name and keywords translation from emojis file and we will use google translation for this, 
`note: I have translated all the name and keywords in my local branch later I will push for testing if we want to`

**2.** 
Next, wherever we are using `emoji.name` for showing the emoji name, now we will use `props.translate(`emojis[${emoji.name}`)`

I am adding a few pages that I came across and we will check the app thoroughly in the PR process where we have to use `props.translate`

1. https://github.com/Expensify/App/blob/d0634b6aa4ec43067467f968c9a7f024c1acad67/src/components/Reactions/QuickEmojiReactions/BaseQuickEmojiReactions.js#L52-L53

2.
https://github.com/Expensify/App/blob/d0634b6aa4ec43067467f968c9a7f024c1acad67/src/components/Reactions/MiniQuickEmojiReactions.js#L64-L68

3. https://github.com/Expensify/App/blob/d0634b6aa4ec43067467f968c9a7f024c1acad67/src/components/Reactions/ReactionTooltipContent.js#L58


**3. Now Comes to the main part here, which is the searching for emoji based on the language preference**

We load all the emojis data in Trie structure here in [EmojiTrie](https://github.com/Expensify/App/blob/main/src/libs/EmojiTrie.js)

It loads one time only when the app loads, so here we will add the NVP_PREFERRED_LOCALE Onyx key to listen to the change, and whenever we change the locale, we will fetch the All emojis in Trie like below

```
Onyx.connect({
    key: ONYXKEYS.NVP_PREFERRED_LOCALE,
    callback: (val) => {
        if (_.isEmpty(val)) {
            return;
        }
        emojisTrie = initializeEmojisTrie();
    },
});
```


here whenever we will add/update data to emojisTrie at that time we will add 1 extra key `localizeName` along with code, types, and suggestions, `localizeName` will be used for display the data and searching the emoji

**const localizeName = Localize.translateLocal(`emojis[${emojis[i].name}]`);**

https://github.com/Expensify/App/blob/0b6168b7c70522be7ad0bd4afcad06060d45dfbc/src/libs/EmojiTrie.js#L14-L18

Here also will add localizeName in the suggestions

https://github.com/Expensify/App/blob/0b6168b7c70522be7ad0bd4afcad06060d45dfbc/src/libs/EmojiTrie.js#L23-L30

Now, then we will need to change a few things in EmojiUtils search mechanism

here we need to check the `nodes[j].metaData.name` insted of `nodes[j].name`

https://github.com/Expensify/App/blob/0b6168b7c70522be7ad0bd4afcad06060d45dfbc/src/libs/EmojiUtils.js#L275


Here we will add `localizeName` to the matching array so that, at the time of showing searched emoji, we will display `localizeName`

https://github.com/Expensify/App/blob/0b6168b7c70522be7ad0bd4afcad06060d45dfbc/src/libs/EmojiUtils.js#L279

`localizeName: nodes[j].metaData.localizeName`

I am attaching a video where I have tried to cover all the common scenarios, please have a look

<details>
  <summary>Result</summary>
  

https://github.com/Expensify/App/assets/35371050/fe670c9e-a17d-409e-a05a-8a49163b2373


</details>



### What alternative solutions did you explore? (Optional)
none

**Comment by s-alves10** [comment]
> @s-alves10 I don't see anything from `Result` section in your proposal. Did you mean to add demo video? I'd like to check your branch with some test Spanish emojis added

This requires some work. If you want to check a demo, sure I'll.

**Comment by stitesExpensify** [comment]
FYI to contributors, "We will use google translate for this" is not an acceptable solution for this problem. We need to use an existing resource that has proper, widely accepted translations. If we use google translate, we will end up with 100 bugs about "this emoji name is not technically correct"

**Comment by iwiznia** [comment]
Did not delve much, but here https://github.com/milesj/emojibase is an emoji library that seems to have translated emojis

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/16086#issuecomment-1545263685)

**Comment by s-alves10** [comment]
@0xmiroslav 
I attached the demo video

**Comment by stitesExpensify** [comment]
Not overdue, @0xmiroslav is testing a proposal. Any update from your end Miroslav?

**Comment by jayeshmangwani** [comment]
Hey @0xmiroslav above proposal selected, or still we are testing?
As I have tested with this emoji [database](https://github.com/Expensify/App/issues/16086#issuecomment-1559950476) and cross-checked every emoji and we can use this database as it has every emoji translated with suggestion keywords we have in our emojis.js database.
if the above proposal will not work then I can push my local branch for testing
The local testing demo video https://drive.google.com/file/d/1MGouzg_6OOBiEqVQyOi29TzBLnAxCEO1/view?usp=sharing

**Comment by s-alves10** [comment]
@0xmiroslav 
What issues are there in my solution?

**Comment by jayeshmangwani** [comment]
> Thanks @jayeshmangwani Other than emoji database, is your solution meaningfully different from @s-alves10?

@0xmiroslav I am unsure how you measure the meaningfully different, but this is my approach. 

1. we have to create a new key in en.js and es.js for emojis and need to use `props.translate('emojis')` instead emojis from the emojis.js file and emojis array has an extra field value, whenever we have to display emoji name then use value and if have to add emoji to API(reaction add/remove) then need to use the name field
2. create a new function `getEmojiLocalizedName` in EmojiUtils for returning emoji value(localized name) based on emoji name
3. We need to use getEmojiLocalizedName for Every file like EmojiSuggestions, MiniQuickEmojiReactions, BaseQuickEmojiReactions, and ReactionTooltipContent where we are displaying emoji name
4. For the emojiTrie we need to listen NVP_PREFERRED_LOCALE key and need to return the emojisTrie based on the locale

**Comment by s-alves10** [comment]
@0xmiroslav 
What standard do we need to meet?
There are 18112 emojis in our original database, and 18111 in my new database. I can diff those and find out which one is missing. 
Name is the same as slack as you can see https://emojiterra.com/es/lista/
![image](https://github.com/Expensify/App/assets/126839717/eb10c0e7-ae6d-412f-8c96-316f51be0fd4)

Not sure keywords should be the same as the old one?
I think we can use the keywords in the new database.

**Comment by s-alves10** [comment]
@0xmiroslav 

We're now using shortcodes and keywords for search. Names are not used, but those are the real meaning of the emoji: name, short code, keywords.

I think storing the emoji data like above would be good. We might need to use name in the future.

Will you kindly tell me why do you think keywords count for each emoji should be the same? I don't think none of the languages has 1:1 matching each other.

**Comment by jayeshmangwani** [comment]
> @jayeshmangwani you suggested to put emoji name in en.js and es.js. How will you localize keywords?
> 

@0xmiroslav We can directly use the emojis array from localized files, English and Spanish will have different emojis array, and here value and keywords under the emoji object will be based on language, code and name will be the same for both languages

My Proposal is outdated, and I have changed a few things at my local branch, if other contributor's Proposals will not work, then I will Update My proposal



>  @jayeshmangwani what is your plan to translate all Spanish emoji names and keywords without any misspelling? We don't guarantee https://github.com/milesj/emojibase or https://emojiterra.com/es/lista/ meet 100% copies to our need.

I have crosschecked every emoji with the emojibase database by comparing the hex code and emojibase has all the emojis translated that we are using in emojis.js on our database

**Comment by s-alves10** [comment]
@0xmiroslav @stitesExpensify 

There are popular chatting application such as GitHub, Slack, Discord, etc. They all support emojis but only Slack supports emoji translation. As you can see in GitHub and Discord, they support English only. I guess current emojis are similar to GitHub but unfortunately it doesn't support Spanish.

How can we measure success in Emoji translation?
People used to use emojis in chatting application and I think we need to follow their words(names, short codes, keywords).  I thought we should do this comparing with Slack because only Slack supports emoji translation.

The [site](https://emojiterra.com/) I've found has Slack translation(short code) and name/keywords translation. And that's why I used this translation library.

Will you share your thoughts regarding the way we measure success in emoji translation and how to implement that.

**Comment by stitesExpensify** [comment]
@s-alves10 I agree that slack is probably the best to compare against. If that website matches then that seems like a good source for names. They don't necessarily need to be 1:1 to our english names, but a full catalog of name and alternative names is definitely required.

With regards to the name, I know that the "real" name is the super long "grinning face with smiling eyes" but I don't think that we need that information since it will never be used.

**Comment by s-alves10** [comment]
@stitesExpensify 

> With regards to the name, I know that the "real" name is the super long "grinning face with smiling eyes" but I don't think that we need that information since it will never be used.

Yes. The real names are not used anywhere now and we can simply remove them. I added them for clarity of the property names.

And I have already extracted all the emojis from the site and they are available here https://github.com/s-alves10/App/blob/fix/issue-16086/assets/new_emojis.js

**Comment by tjferriss** [comment]
@stitesExpensify @0xmiroslav what is the next step here? Are we comfortable with @s-alves10 proposal?

**Comment by stitesExpensify** [comment]
I am comfortable with @s-alves10 proposal. The only thing I want to change is `Create 2 versions of EmojiTree in EmojiTrie.js.`. We should connect to onyx and only create 1 trie based on the current language instead of 2. Otherwise when we eventually add more languages we'll end up using a lot of memory unnecessarily.

**Comment by s-alves10** [comment]
I'll create a PR in a day or two

**Comment by s-alves10** [comment]
@0xmiroslav 

PR is ready for review
I've verified that all emojis have en and es name

**Comment by s-alves10** [comment]
@0xmiroslav 

Did you get a chance to review the [PR](https://github.com/Expensify/App/pull/20828)?

**Comment by stitesExpensify** [comment]
At one point we intentionally decided to not have a people and body category. It was combined into the `smileysAndEmotion` section, and should not be re-added

**Comment by s-alves10** [comment]
Okay

**Comment by s-alves10** [comment]
@0xmiroslav 
People and body category removed. Spanish version of category names updated to the original version.

**Comment by s-alves10** [comment]
@0xmiroslav @stitesExpensify 

Please check https://github.com/Expensify/App/pull/20828#issuecomment-1595184585 and let me know your thoughts.
I know minimum changes are required but intentionally made some changes for future improvements.
If you still don't like them, sure, I'll fix them.

Thank you

**Comment by iwiznia** [comment]
I am here to make a small request: can we make it so that emojis auto complete always in the language you have set AND in english?

**Comment by s-alves10** [comment]
I can see slack is doing so.

**Comment by stitesExpensify** [comment]
@iwiznia just to make sure I understand, you're asking that we would always create 2 tries and search them both? So if we have english, spanish, german, russian etc. we would always build the English Trie, and then only one other (the one you have set) and search that too. We would not search all languages every time. Is that right?

**Comment by s-alves10** [comment]
@stitesExpensify 

Yes. Slack is working that way

Did you get a chance to take a look at https://github.com/Expensify/App/pull/20828#issuecomment-1596329332?

**Comment by iwiznia** [comment]
Yes, that's correct. My reasoning is that lots of people (including me) know the emoji names in english, but not in spanish. This is the same that slack does (and I think Whatsapp too)

**Comment by stitesExpensify** [comment]
@iwiznia what does slack do for the "xx reacted with :name:"? Does is display in english or spanish?

**Comment by iwiznia** [comment]
I don't know what slack does, but in newDot everything should be shown in the language of your choice.

**Comment by stitesExpensify** [comment]
So even if you typed `:smile:` it should show `:sonrisa:`?

**Comment by iwiznia** [comment]
IMO yes. The "match also english" is so that people don't need to learn the emoji names in both languages. After posting it, everything should display in your preferred language.

**Comment by stitesExpensify** [comment]
Makes sense to me!

**Comment by tjferriss** [comment]
It looks like there is some discussion on the actual PR. @stitesExpensify now that you're back from OOO can you chime in when you get a chance?

**Comment by stitesExpensify** [comment]
Commented on the PR, it's looking great! If anyone (preferably spanish speakers) wants to test it out now's the time!

**Comment by stitesExpensify** [comment]
So here's what I'm thinking:
1. This is new functionality, so technically it can work however we want (nobody is trying to search with spanish names right now)
2. People are very busy and our spanish speakers don't have time to test right now (they also don't really have an obligation to)

Let's just test it the best that we can, make sure we like the code, and get this released!

**Comment by tjferriss** [comment]
@0xmiroslav and @s-alves10 I've sent an offer to both of you on Upworks for payment: https://www.upwork.com/jobs/~0194e3070a01bf0b21

**Comment by tjferriss** [comment]
All set for payments on the 20th

**Comment by tjferriss** [comment]
@s-alves10 your payment has been sent.

@0xmiroslav I've held off on your payment and moved the issue to monthly. Please let me know when you're ready for payment.

**Comment by tjferriss** [comment]
The payment has been sent via Upworks.

**Comment by tjferriss** [comment]
I'm sorry @0xmiroslav, I did not read the Slack thread. I assumed it was ok to proceed with regular payment based on your comment. In the future, I would specifically call out the Upworks account that needs to be paid out in the comment.
